internal/controller/compute/user/user.go

```diff
@@ -0,0 +1,325 @@
+		}
+	}
+
+	setStatus(cr, observed, cr.Spec.ForProvider.GroupIDs)
+
+	return managed.ExternalCreation{ConnectionDetails: conn}, nil
+}
+
+func (eu *externalUser) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mg.(*v1alpha1.User)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.Wrap(errors.New(errNotUser), "update error")
+	}
+
+	userID := cr.Status.AtProvider.UserID
+
+	observed, resp, err := eu.service.UpdateUser(ctx, userID, cr.Spec.ForProvider)
+	if err != nil {
+		werr := errors.Wrap(err, errUserUpdate)
+		return managed.ExternalUpdate{}, compute.AddAPIResponseInfo(resp, werr)
+	}
+	conn := connectionDetails(cr, observed)
+
+	err = updateGroups(ctx, eu, userID, cr.Status.AtProvider.GroupIDs, cr.Spec.ForProvider.GroupIDs)
+	if err != nil {
+		return managed.ExternalUpdate{ConnectionDetails: conn}, errors.Wrap(err, "failed to update groups")
+	}
+
+	return managed.ExternalUpdate{ConnectionDetails: conn}, nil
+}
+
+func (eu *externalUser) Delete(ctx context.Context, mg resource.Managed) error {
+	user, ok := mg.(*v1alpha1.User)
+	if !ok {
+		return errors.Wrap(errors.New(errNotUser), "delete error")
+	}
+	user.SetConditions(xpv1.Deleting())
+
+	userID := user.Status.AtProvider.UserID
+
+	for _, groupID := range user.Status.AtProvider.GroupIDs {
+		if err := eu.service.DeleteUserFromGroup(ctx, groupID, userID); err != nil {
+			return errors.Wrap(err, "failed to remove user from a group")
+		}
+	}
+
+	resp, err := eu.service.DeleteUser(ctx, userID)
+	return compute.ErrorUnlessNotFound(resp, errors.Wrap(err, errUserDelete))
+}
+
+// updateGroups adds or remove groups for the userID.
+func updateGroups(ctx context.Context, eu *externalUser, userID string, atProviderGroups []string, forProviderGroups []string) error {
+	for _, groupID := range atProviderGroups {
+		if !slices.Contains(forProviderGroups, groupID) && groupID != "" {
+			if derr := eu.service.DeleteUserFromGroup(ctx, groupID, userID); derr != nil {
+				return errors.Wrap(derr, "failed to remove user from a group")
+			}
+		}
+	}
+
+	for _, groupID := range forProviderGroups {
+		_, _, gerr := eu.service.AddUserToGroup(ctx, groupID, userID)
+		// skip if user is already member of this group.
+		if gerr != nil && strings.Contains(gerr.Error(), "is already member of") {
+			continue
+		}
+		if gerr != nil {
+			return errors.Wrap(gerr, errAddUserToGroup)
+		}
+	}
+	return nil
+}
+
+// isUserUpToDate returns true if the User is up-to-date or false otherwise.
+func isUserUpToDate(params v1alpha1.UserParameters, observed ionosdk.User, observedGroups []string) bool { //nolint:gocyclo
+	if !observed.HasProperties() {
+		return false
+	}
+
+	// After creation the password is stored as a connection detail secret
+	// and removed from the cr. If the cr has a password it means
+	// the client wants to update it.
+	if params.Password != "" {
+		return false
+	}
+
+	if !reflect.DeepEqual(params.GroupIDs, observedGroups) {
```

Comment:
if the items are in different orders it will think they are not equal. I did simple verification to verify it.
```suggestion
ipackage main

import (
	"fmt"
	"reflect"
)

func main() {
	first := []string{"100", "200", "300"}
	second := []string{"300", "100", "200"}
	third := []string{"100", "200", "300"}

	if !reflect.DeepEqual(first, second) {
		fmt.Println("not equal")
	} else {
		fmt.Println("equal")
	}

	if !reflect.DeepEqual(first, third) {
		fmt.Println("not equal")
	} else {
		fmt.Println("equal")
	}
}
```